src/cmds/scripts/win_postinstall.py

```diff
@@ -0,0 +1,632 @@
+import sys
+import getopt
+import os
+import subprocess
+import socket
+from tempfile import mkstemp
+from shutil import move
+import time
+
+install_path = ''
+pbs_path = ''
+
+
+def replace(file_path, pattern, subst):
+    fh, abs_path = mkstemp()
+    with os.fdopen(fh, 'w') as new_file:
+        with open(file_path) as old_file:
+            for line in old_file:
+                new_file.write(line.replace(pattern, subst))
+    # Remove original file
+    os.remove(file_path)
+    # Move new file
+    move(abs_path, file_path)
+
+
+def ValidateCredentials(username, password):
+    if (username != '' and password != ''):
```

Comment:
The idea being that the precedence rules are arcane enough that nobody should be expected to remember them all, neither the original programmer nor the maintenance programmer reading the code, so it is better to make it explicit. 
 Essentially the parentheses serve both to communicate the intent to the compiler and as documentation for the next someone who has to work on it.
I believe in Python those two statements will generate the same bytecode so you're not even losing any efficiency.
Link: https://stackoverflow.com/questions/4740419/parentheses-in-python-conditionals